packages/frontend/core/src/components/pure/cmdk/highlight.tsx

```diff
@@ -0,0 +1,80 @@
+import { memo } from 'react';
+
+import {
+  highlightContainer,
+  highlightKeyword,
+  highlightText,
+  labelContent,
+  labelTitle,
+} from './highlight.css';
+
+type SearchResultLabel = {
+  title: string;
+  subTitle?: string;
+};
+
+type HighlightProps = {
+  text: string;
+  highlight: string;
+};
+
+type HighlightLabelProps = {
+  label: SearchResultLabel;
+  highlight: string;
+};
+
+function escapeRegExp(string: string) {
+  return string.replace(/[.*+?^${}()|[\]\\]/g, '\\$&');
+}
+
+export const Highlight = memo(function Highlight({
+  text = '',
+  highlight = '',
+}: HighlightProps) {
+  const regex = highlight.trim()
+    ? new RegExp(`(${escapeRegExp(highlight)})`, 'ig')
+    : null;
+
+  if (!regex) {
+    return <span>{text}</span>;
+  }
+  const parts = text.split(regex);
```

Comment:
not sure why use regex, not jus split by string?

Reply:
Regex is used to ignore case